rfcs/56-better-reflect.md

```diff
@@ -0,0 +1,261 @@
+# bevy_reflect: Unique `Reflect`
+
+This builds on https://github.com/bevyengine/bevy/pull/4042 and the `TypeInfo`
+API.
+
+## Summary
+
+Add a `ReflectProxy` enum that generalizes the `Dynamic***` structs from
+`bevy_reflect` to all types, and remove `Reflect` implementation for
+`Dynamic***` structs.
+
+I also suggest renaming all `Dynamic***` into `***Proxy` and the reflect
+traits `clone_dynamic` methods to `proxy_clone`.
```

Comment:
I like the idea of a rename. The one potential issue with it is that it's not inherently true. 

You can use `DynamicStruct` without ever needing to convert to a "proxied type". It can just be a tool to use in general and on its own apart from proxying.